### ISSUE TITLE
please add modules to work with libraries

### ISSUE THREAD

**Issue by RobVerduijn** [ansible-collections/vmware.vmware_rest#149]
##### SUMMARY

Please add modules to work with libraries.

There are no modules to work with libraries
- create a library item (template) from a vm
- create a vm from a library item (template)
- upload an item to a library
- download an item from a library
- rename items in libraries 

##### ISSUE TYPE
- Feature Idea

##### ADDITIONAL INFORMATION

```yaml

```

**Comment by Akasurde** [comment]
I will work on this.

**Comment by timblaktu** [comment]
Thanks for working on this @Akasurde. I see in [your PR #79](https://github.com/ansible-collections/vmware_rest_code_generator/pull/79) you made the changes to the code generation project to support content libraries, but I would benefit from an update here on the scope of remaining changes before you've generated and committed the resulting modules to `vmware_rest` project. I'm specifically interested in those  that will enable working with templates, as requested in the closed issues:
* https://github.com/ansible-collections/vmware.vmware_rest/issues/165
* https://github.com/ansible-collections/vmware.vmware_rest/issues/148

Thanks!

**Comment by goneri** [comment]
I close this feature request because we've regenerated the collection and this time and it includes the new library modules. They will be released with the 2.0.0 release.

**Comment by RobVerduijn** [comment]
awesome

**Comment by timblaktu** [comment]
Cool, thanks for the update. I've moved on from using ansible to create infrastructure, and am now separating infrastructure creation from configuration management, and am using Packer + Pulumi for Infrastructure creation, and ansible ONLY for configuration now. I'm much happier with this arrangement. But I'm curious in what context people are using ansible and the various vmware collections for infrastructure creation, and finding it valuable? 

I was able to do in one day with Packer what I had tried to do in 2 months with ansible's various vmware collections. I'm not trying to detract from the project, @goneri is doing awesome work here, I'm just really curious what I may be missing, and what you see as the future of this collection and gaining adoption for using ansible for infrastructure.
